### PR TITLE
Removed  deprecated methods. Updated structure to use separated client configs.

### PR DESCRIPTION
**Pull request by miry**
When we have multiple HasOffers clients with different API token settings and we use a singleton configuration, it's possible that several sidekiq workers will overwrite each other's configuration.

Changed structure to use separate client for each configuration.

### CODE REVIEW

File: lib/hasoffersv3/base.rb [hunk 1]
```diff
@@ -3,43 +3,11 @@
...
-      def requires!(hash, required_params)
-        missing_params = []
-        required_params.each do |param|
-          missing_params.push param unless hash.has_key?(param)
-        end
-        unless missing_params.empty?
-          raise ArgumentError.new("Missing required parameter(s): #{missing_params.join(', ')}")
-        end
-      end
+    def initialize(client=HasOffersV3.client)
```
Comment (psychocandy) [inline lib/hasoffersv3/base.rb]: `def initialize(client = HasOffersV3.client)`

File: lib/hasoffersv3/base.rb [hunk 2]
```diff
@@ -48,16 +16,63 @@ def client
         HasOffersV3.client
       end
 
+      def method_missing(meth, *args, &block)
```
Comment (psychocandy) [inline lib/hasoffersv3/base.rb]: Isn't it better to raise in this case?

Comment (miry) [inline lib/hasoffersv3/base.rb]: I added this method to support old way to use HO for backward compatible.

File: lib/hasoffersv3/base.rb (hunk 1 above)
Comment (miry) [inline lib/hasoffersv3/base.rb]: fixed

**Comment by psychocandy** [comment]
:+1:

**Pull request merged**